File: C++/1.자료구조/hash/문자열집합14425.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
  int n,m;
  cin >> n >> m;
  unordered_map<string,int> map1;
  int answer = 0;
  for(int i=0;i<n;i++) {
    string a;
    cin >> a;
    map1[a]++;
  }
  for(int i=0;i<m;i++) {
    string a;
    cin >> a;
    if(map1[a]!=0) answer++; 
  }
  cout << answer << '\n';
  return 0;
}

File: Java/1.자료구조/hash/문자열집합14425.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.StringTokenizer;

public class Main {
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		StringTokenizer st;
		HashMap<String, Integer> map1 = new HashMap<String,Integer>();
		
		st = new StringTokenizer(br.readLine()," ");
		int n = Integer.parseInt(st.nextToken());
		int m = Integer.parseInt(st.nextToken());
		for(int i=0;i<n;i++) {
			String x = br.readLine();
			map1.put(x, 1);
		}
		int answer = 0;
		for(int i=0;i<m;i++) {
			String x = br.readLine();
			if(map1.get(x)!=null) {
				answer+=1;
			}
		}
		System.out.println(sb.append(answer));
	}
}
